src/ol_infrastructure/components/aws/olvpc.py

```diff
@@ -80,6 +81,30 @@ def is_private_net(cls: "OLVPCConfig", network: IPv4Network) -> IPv4Network:
             )
         return network
 
+    @validator("k8s_service_subnet")
+    def k8s_service_subnet_is_subnet(
+        cls: "OLVPCConfig", k8s_service_subnet: Optional[IPv4Network], values: Dict
+    ) -> Optional[IPv4Network]:
+        """Ensure that specified k8s subnet is actually a subnet of the cidr specified for the VPC.
+
+        :param k8s_service_subnet: The K8S service subnet to be created in the VPC.
+        :type k8s_service_subnet: IPv4Network
+
+        :param values: Dictonary containing the rest of the class values
+        :type values: Dict
+
+        :raises ValueError: Raise a ValueError if the specified subnet is not actually a subnet of the VPC cidr
+
+        :returns: The K8S service subnet
+
+        :rtype: IPv4Network
+        """
+        network = values.get("cidr_block")  # type: ignore
+        assert network is not None
+        if k8s_service_subnet is not None and not network.overlaps(k8s_service_subnet):
```

Comment:
The `overlaps` method isn't necessarily an accurate representation of what we want, since it doesn't guarantee that it is a full subnet, just that there is _some_ overlap in the ranges. It might be more accurate and semantically descriptive to use [`subnet_of`](https://docs.python.org/3/library/ipaddress.html#ipaddress.IPv4Network.subnet_of)

Reply:
Smort.